code/modules/mob/living/carbon/carbon.dm

```diff
@@ -315,6 +315,9 @@
 	if(.)
 		if(visual)
 			return
+		if(weakeyes)
```

Comment:
Well, yes and no---mostly yes though, because the biggest users of weakeyes are lings---who will still have their thermal "eyes" even if you rip out their primary eyes.